### ISSUE TITLE
Classic editor's voice label is semantically incorrect

### ISSUE THREAD

**Issue by oleq** [ckeditor/ckeditor5#15352]
## 📝 Provide a description of the improvement

The classic editor's voice label that we have is not used properly (`<label id="foo"> + <div class="ck-editor" aria-labelledby="foo">`). For instance, the label is missing the `for` attribute. And even if it had it, [this would still be incorrect](https://validator.w3.org/nu/?showsource=yes&doc=data%3Atext%2Fhtml%3Bcharset%3Dutf-8%3Bbase64%2CPCFET0NUWVBFIGh0bWw%2BDQo8aHRtbCBsYW5nPSIiPg0KPGhlYWQ%2BDQo8dGl0bGU%2BVGVzdDwvdGl0bGU%2BDQo8L2hlYWQ%2BDQo8Ym9keT4NCjxsYWJlbCBmb3I9ImZvbyI%2BTGFiZWw8L2xhYmVsPiA8ZGl2IGlkPSJmb28iPjwvZGl2Pg0KPC9ib2R5Pg0KPC9odG1sPg%3D%3D).

On top of it:
* `<label for="foo"> + <div class="ck-editor" id="foo">` combo, besides also being semantically incorrect, does not get recognized by screen readers when the label is hidden anyway (NVDA)
    * there's no place we can display it in the classic editor's UI.
* having the `for` attribute would mean we would need to assign ids to editors, which is a complex topic.

## Possible solutions

* Use `aria-label` (needs to be checked),
* Rename label to span `<label id="foo"> + <div class="ck-editor" aria-labelledby="foo">` -> `<span id="foo"> + <div class="ck-editor" aria-labelledby="foo">`
---

If you'd like to see this improvement implemented, add a 👍 reaction to this post.
